### PR TITLE
Update and try fix GH Pages publish action.

### PR DESCRIPTION
**Pull request by tristanmorgan**
from the latest version of the [gh-pages docs](https://github.com/peaceiris/actions-gh-pages/blob/main/README.md#%EF%B8%8F-set-runners-access-token-github_token) it should use GITHUB_TOKEN to authenticate...

### CODE REVIEW

**Comment by tristanmorgan** [comment]
@aaronhurt tested on my fork [see](https://github.com/tristanmorgan/fabio/actions/runs/10763933663/job/29846039987)

File: .github/workflows/github-pages.yml [hunk 1]
```diff
@@ -1,24 +1,22 @@
...
-      run: |
-        export PATH=${PATH}:${HOME}/bin
-        make github-pages
-    - name: deploy
-      uses: peaceiris/actions-gh-pages@v3
-      with:
-        github_token: ${{ secrets.GH_ACTIONS_TOKEN }}
-        publish_dir: ./docs/public
-        cname: fabiolb.net
+      - uses: actions/checkout@master
```
Comment (dcarbone) [inline .github/workflows/github-pages.yml:10]: Change this to `@v4`, please

**Review by dcarbone — CHANGES_REQUESTED** [review]
One small change, otherwise lgtm

**Review by nathanejohnson — CHANGES_REQUESTED** [review]
it looks like there are whitespace issues with this commit as well, the file uses spaces for indent

**Comment by tristanmorgan** [comment]
I pinned the actions/checkout version to v4, and dropped the formatting commit.

**Review by nathanejohnson — APPROVED** [review]

**Review by dcarbone — APPROVED** [review]

**Pull request merged**